nifi-docker/dockerhub/Dockerfile

```diff
@@ -16,37 +16,37 @@
 # under the License.
 #
 
-FROM openjdk:8
-MAINTAINER Apache NiFi <dev@nifi.apache.org>
+FROM openjdk:8-jre
+LABEL maintainer "Apache NiFi <dev@nifi.apache.org>"
 
 ARG UID=1000
-ARG GID=50
 ARG NIFI_VERSION=1.4.0
+ARG MIRROR=https://archive.apache.org/dist
 
-ENV NIFI_BASE_DIR /opt/nifi
-ENV NIFI_HOME $NIFI_BASE_DIR/nifi-$NIFI_VERSION
-ENV NIFI_BINARY_URL https://archive.apache.org/dist/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
+ENV NIFI_BASE_DIR /opt/nifi 
+ENV NIFI_HOME=$NIFI_BASE_DIR/nifi-$NIFI_VERSION \
+    NIFI_BINARY_URL=/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
 
 # Setup NiFi user
-RUN groupadd -g $GID nifi || groupmod -n nifi `getent group $GID | cut -d: -f1`
-RUN useradd --shell /bin/bash -u $UID -g $GID -m nifi
-RUN mkdir -p $NIFI_HOME
+RUN groupadd nifi && useradd nifi --shell /bin/bash -u $UID -m -g nifi  \
```

Comment:
I'm just saying, if we're going to remove $GID, we should remove it entirely.  But if we're going to support it (i.e. by having an option in the Dockerbuild.sh), then we should support it entirely.  Line 31 currently ignores $GID.